include/splinterdb/data.h

```diff
@@ -25,10 +25,11 @@
 #include "splinterdb/public_util.h"
 
 typedef enum message_type {
+   MESSAGE_TYPE_INVALID = 0,
    MESSAGE_TYPE_INSERT,
    MESSAGE_TYPE_UPDATE,
    MESSAGE_TYPE_DELETE,
-   MESSAGE_TYPE_INVALID,
+   MESSAGE_TYPE_MAX_VALID_USER_TYPE = MESSAGE_TYPE_DELETE,
 } message_type;
```

Comment:
Message types are written to disk as well, so this change will also cause a backward compatibility issue.

Reply:
@ajhconway - Yes. We understand that this change will cause a backward compatibility issue. But, as Splinter is at early version, the expectation is that this change will be acceptable.

@rtjohnso I added a new message type as you requested. And, to distinguish message types for internal functions from ones for users (which are disk-resident), the message type will start with a value of 1000. 

And I added a comment for this enum type. Please review that and let me know if you need any correction.